app/models/user.rb

```diff
@@ -3,6 +3,12 @@ class User < ApplicationRecord
   attr_accessor :remember_token, :activation_token, :reset_token
 
   has_many :microposts, dependent: :destroy
+  has_many :active_relationships, class_name: Relationship.name,
+    foreign_key: :follower_id, dependent: :destroy
+  has_many :passive_relationships, class_name: Relationship.name,
+    foreign_key: :followed_id, dependent: :destroy
+  has_many :following, through: :active_relationships, source: :followed
+  has_many :followers, through: :passive_relationships, source: :follower
```

Comment:
:active_relationship là mối quan hệ "follow"
:passive_relationship là mối quan hệ "được follow bởi"
2 cái này đều lấy từ class Relationship nên cái tên của nó Relationship.name (giống kiểu to string)

active relationship có khóa ngoài là follower_id tham chiếu lên bảng User
passive relationship có khóa ngoài followed_id tham chiếu lên bảng User

dependent: :destroy có nghĩa là nếu trên bảng User một follower bị xóa thì các active_relationships tương ứng với follower_id của nó cũng bị xóa theo
tương tự với dependent: : destroy của passive relationships 
3. has_many :following, through: :active_relationships, source: :followed
có nghĩa là mỗi following ở đây chính là một followed trong active_relationship 
cái này để gọi cho dễ, vì gọi followeds rất dở
4. tương tự cái số 3